pkg/api/validation/validation.go

```diff
@@ -2348,9 +2348,16 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) field.ErrorList {
 		activeDeadlineSeconds := *oldPod.Spec.ActiveDeadlineSeconds
 		mungedPod.Spec.ActiveDeadlineSeconds = &activeDeadlineSeconds
 	}
+
+	// Allow tolerations updates.
+	if len(newPod.Spec.Tolerations) > 0 {
+		allErrs = append(allErrs, validateTolerations(newPod.Spec.Tolerations, fldPath.Child("tolerations"))...)
+	}
+	mungedPod.Spec.Tolerations = oldPod.Spec.Tolerations
```

Comment:
Technically the thing we agreed on is that you should only be able to add tolerations, not change or delete existing ones. (Though I'm not sure the reasoning still holds.) Anyway, probably easiest thing to do is to add a function (and call it from here) like
```
validateOnlyAddedTolerations(newTolerations []api.Toleration,
           oldTolerations []api.Toleration, , fldPath *field.Path) field.ErrorList {
  // if any <Key,Operator,Value,Effect> is in oldTolerations but not in
  // newTolerations, then give error "toleration update can only add
  // new tolerations, not delete or modify existing ones."
}
```